File: Week1/simple/旋转数组/Solution.java
package 旋转数组;
import java.util.Arrays;

/*
执行用时：0 ms, 在所有 Java 提交中击败了100.00%的用户
内存消耗：39 MB, 在所有 Java 提交中击败了82.53%的用户
* */

public class Solution {
    public static void main(String[] args) {
        int[] nums = {1,2,3,4,5,6,7};
        int k = 3;
        rotate(nums, k);
    }
    public static void rotate(int[] nums, int k) {
        int _k = k % nums.length;
        if (_k == 0) return;
        reverse(nums, 0, nums.length - 1);
        reverse(nums, 0, _k - 1);
        reverse(nums, _k, nums.length - 1);
        System.out.println(Arrays.toString(nums));
    }
    public static void reverse(int[] nums, int start, int end){
        while (start < end){
            int temp = nums[end];
            nums[end] = nums[start];
            nums[start] = temp;
            start++;
            end--;
        }
    }
}

File: Week1/simple/旋转数组/Solution.js
/*
执行用时：84 ms, 在所有 JavaScript 提交中击败了97.61%的用户
内存消耗：38.4 MB, 在所有 JavaScript 提交中击败了42.93%的用户
* */

/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
    let _k = k % nums.length;
    if (_k === 0) return;
    reverse(nums)
    reverse(nums,0,k-1)
    reverse(nums,k)
};
function reverse(list=[],start=0,end=list.length-1) {
    let s = start;
    let e = end;
    while (s < e){
        let temp = list[e];
        list[e] = list[s];
        list[s] = temp;
        s++;
        e--;
    }
}
